src/reify.ml4

```diff
@@ -172,6 +197,16 @@ struct
     in
     go (String.length s - 1) tEmptyString
 
+  let string_hash = Hashtbl.create 420
+
+  let quote_string s =
+    if !share_strings then
+      try Hashtbl.find string_hash s
+      with Not_found ->
+        let qs = quote_string s in
+        Hashtbl.add string_hash s qs; qs
+    else quote_string s
```

Comment:
Not so much I think, at least naturally we would share suffixes not prefixes due to the inductive nature of strings in Coq.